crates/tabby-common/src/api/event.rs

```diff
@@ -49,6 +49,17 @@ pub enum Event {
         #[serde(skip_serializing_if = "Option::is_none")]
         user: Option<String>,
     },
+    ChatCompletion {
+        completion_id: String,
+        input: Vec<Message>,
+        output: Vec<String>,
```

Comment:
output format should be `Message`, as precious comment suggested

Reply:
The only extra data associated with that is the role, which is not generated by the stream. There are multiple input messages, each which holds its own role. Will they ever differ, or are they all the same? If they're all the same, I can just store the first role that is received and associate it with each output chunk.